spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/LocalizedQueueConnectionFactory.java

```diff
@@ -364,6 +364,9 @@ public void destroy() throws Exception {
 		for (ConnectionFactory connectionFactory : this.nodeFactories.values()) {
 			((DisposableBean) connectionFactory).destroy();
```

Comment:
I think for safety and clear code we have to use ` instanceof DisposableBean` here as well